query-algebrizer/src/clauses/mod.rs

```diff
@@ -191,6 +218,38 @@ impl Default for ConjoiningClauses {
     }
 }
 
+/// Cloning.
+impl ConjoiningClauses {
+    fn make_receptacle(&self) -> ConjoiningClauses {
+        let mut concrete = ConjoiningClauses::default();
+        concrete.is_known_empty = self.is_known_empty;
+        concrete.empty_because = self.empty_because.clone();
+
+        concrete.input_variables = self.input_variables.clone();
+        concrete.value_bindings = self.value_bindings.clone();
+        concrete.known_types = self.known_types.clone();
+        concrete.extracted_types = self.extracted_types.clone();
+
+        concrete
+    }
+
+    /// Make a new CC populated with the relevant variable associations in this CC.
+    /// Note that the CC's table aliaser is not yet usable. That's not a problem for templating for
+    /// simple `or`.
```

Comment:
Is this still valid after your later commits?  Make sure the commit agrees with the final reality.